sources/packages/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -203,4 +209,78 @@ export class ApplicationControllerService {
       );
     }
   }
+
+  /**
+   * Process application income verification details.
+   * @param CRAIncomeVerification CRA income verification data.
+   * @return processed object.
+   */
+
+  processApplicationIncomeVerificationDetails(
+    CRAIncomeVerification: CRAIncomeVerification[],
+  ): ApplicationIncomeVerification {
+    const incomeVerificationDetails = {} as ApplicationIncomeVerification;
+    const studentIncomeVerification = CRAIncomeVerification.filter(
+      (incomeVerification) => !incomeVerification.supportingUser,
+    );
+    const supportingUserIncomeVerification = CRAIncomeVerification.filter(
+      (incomeVerification) => incomeVerification.supportingUser,
+    );
+    studentIncomeVerification.forEach((incomeVerification) => {
+      // Student income verification details.
+      incomeVerificationDetails.studentIncomeVerificationStatus = {
+        waiting: !incomeVerification.dateReceived,
+        success: !!incomeVerification.dateReceived,
+      };
+    });
```

Comment:
Trying to understand what is happening here. Is the intention to create an array ?

Reply:
i will look something like 
![image](https://user-images.githubusercontent.com/77353155/192638757-c6f63557-aa21-440d-ba21-9956c1023a71.png)